File: app/src/main/java/io/github/sithengineer/motoqueiro/app/Preferences.java
package io.github.sithengineer.motoqueiro.app;

import android.content.SharedPreferences;

public class Preferences {
  private static final String MI_BAND_ADDRESS = "mi_band_address";
  private static final String RIDE_ID = "ride_id";

  private final SharedPreferences preferences;

  public Preferences(SharedPreferences preferences) {
    this.preferences = preferences;
  }

  public String getMiBandAddressOrDefault() {
    return preferences.getString(MI_BAND_ADDRESS, "");
  }

  public void setMiBandAddress(String miBandAddress) {
    preferences.edit().putString(MI_BAND_ADDRESS, miBandAddress).apply();
  }

  public String getRideId() {
    return preferences.getString(RIDE_ID, null);
  }

  public void setRideId(String rideId) {
    preferences.edit().putString(RIDE_ID, rideId).apply();
  }
}

File: app/src/main/java/io/github/sithengineer/motoqueiro/app/Preferences.kt
package io.github.sithengineer.motoqueiro.app

import android.content.SharedPreferences

class Preferences(private val preferences: SharedPreferences) {

  var miBandAddress: String
    get() = preferences.getString(MI_BAND_ADDRESS, "")
    set(newMiBandAddress) = preferences.edit().putString(MI_BAND_ADDRESS, newMiBandAddress).apply()

  var rideId: Long
    get() = preferences.getLong(RIDE_ID, 0)
    set(rideId) = preferences.edit().putLong(RIDE_ID, rideId).apply()

  companion object {
    private const val MI_BAND_ADDRESS = "mi_band_address"
    private const val RIDE_ID = "ride_id"
  }
}
